pandas/core/indexes/interval.py

```diff
@@ -1003,6 +1004,63 @@ def get_loc(self, key, method=None):
             else:
                 return self._engine.get_loc(key)
 
+    def get_loc_exact(self, key, method=None):
+        """Get integer location, slice or boolean mask for exact
+        Interval matches only.
+
+        Parameters
+        ----------
+        key : Interval
+            The label we want to find locations for. Must have type
+            :class:`Interval`
+        method : {None}, optional
+            * default: matches where the label exactly matches a given
+              :class:`Interval`.
+
+        Returns
+        -------
+        loc : int if unique index, slice if monotonic index, else mask
+
+        Raises
+        ------
+        KeyError if key is not found
+
+        Examples
+        ---------
+        >>> i1, i2 = pd.Interval(0, 1), pd.Interval(1, 2)
+        >>> index = pd.IntervalIndex([i1, i2])
+        >>> index.get_loc_exact(i1)
+        0
+
+        If an exact match is not found, a KeyError is raised
+
+        >>> index.get_loc_exact(pd.Interval(0.5, 1.5))
+        KeyError: Interval(0.5, 1.5, closed='right')
+
+        If a label is in several locations, you get all the relevant
+        locations.
+
+        >>> index = pd.IntervalIndex([i1, i2, i1])
+        >>> index.get_loc_exact(i1)
+        array([0, 2], dtype=int64)
+
+        See Also
+        --------
+        get_loc
+        """
+        all_matches = self.get_loc(key, method=method)
+        if is_scalar(all_matches):
+            if self[all_matches] == key:
+                return all_matches
+            raise KeyError(key)
+        else:
+            exact_matches = self[all_matches] == key
+            if (exact_matches).all():
+                return all_matches
+            elif (exact_matches).any():
```

Comment:
this can also be simplified i think

Reply:
I don't think so: in the ``exact_matches.all()`` case, all_matches might be a slice object, so ``all_matches[exact_matches]`` would then fail.